tsl/src/remote/dist_commands.c

```diff
@@ -74,18 +74,21 @@ ts_dist_cmd_collect_responses(List *requests)
 }
 
 /*
- * Invoke a SQL statement (command) on the given data nodes.
+ * Invoke multiple SQL statements (commands) on the given data nodes.
  *
  * The list of data nodes can either be a list of data node names, or foreign
  * server OIDs.
  *
  * If "transactional" is false then it means that the SQL should be executed
  * in autocommit (implicit statement level commit) mode without the need for
- * an explicit 2PC from the access node
+ * an explicit 2PC from the access node.
+ * 
+ * If "multiple_cmds" is false then "sql" and "params" are not iterated over.
  */
 DistCmdResult *
-ts_dist_cmd_params_invoke_on_data_nodes(const char *sql, StmtParams *params, List *data_nodes,
-										bool transactional)
+ts_dist_multi_cmds_params_invoke_on_data_nodes(const char **sql, StmtParams **params,
+											   List *data_nodes, bool multiple_cmds,
+											   bool transactional)
```

Comment:
I think it might be better to make this a little more generic, but we can take that in a separate commit.